hris_publisher/task.py

```diff
@@ -0,0 +1,94 @@
+import logging
+
+from cis.publisher import ChangeDelegate
+from cis.settings import get_config
+
+import pprint
+
+logger = logging.getLogger(__name__)
+
+
+class CISTask(object):
+    def __init__(self, boto_session, vault_record, hris_groups):
+        self.boto_session = boto_session
+        self.vault_record = vault_record
+        self.hris_groups = hris_groups
+
+    @property
+    def publisher(self):
+        config = get_config()
+        return {
+            'id': config('publisher_name', namespace='cis')
+        }
+
+    def prep(self):
+        current_group_list = self.get_groups_for_record()
+        current_group_list = self.clean_hris_assertions(current_group_list)
+        self.reintegrate_groups(current_group_list)
+        return self.construct_profile()
+
+    def send(self, data):
+        cis_change = ChangeDelegate(self.publisher, {}, data)
+        cis_change.boto_session = self.boto_session
+
+        result = cis_change.send()
+        logger.info('Result of the change for user: {user} is {result}'.format(
+                user=self.vault_record.get('primaryEmail'),
+                result=result
+            )
+        )
+        return result
+
+    def get_groups_for_record(self):
+        return self.vault_record.get('groups')
+
+    def clean_hris_assertions(self, current_group_list):
+        for group in current_group_list:
+            # Remove existing groups with hris_ attrs from assertions.
+            if group.find('hris_') == 0:
+                logger.info('Removing current hris_ assertion from {user} for group : {group}'.format(
+                        user=self.vault_record.get('primaryEmail'),
+                        group=group
+                    )
+                )
+                current_group_list.pop(current_group_list.index(group))
+        return current_group_list
+
+    def reintegrate_groups(self, current_group_list):
+        # Add the new assumptions we made about the user to the current grouplist
+        self.vault_record['groups'] = current_group_list + self.hris_groups
+        logger.info('Groups reintegrated for user: {user}'.format(user=self.vault_record.get('primaryEmail')))
+
+    def construct_profile(self):
+            logger.info('Sending profile to CIS for: {user}'.format(user=self.vault_record.get('primaryEmail')))
+
+            data = {
+                'user_id': self.vault_record.get('user_id'),
```

Comment:
reconstructing the profile seems operationally dangerous if we're able to avoid it - compared to loading the full profile
besides, i believe this should call the CIS Person API in some future were we actually pay tech debt.
Just leaving this as a comment for now